src/main/java/com/algorand/algosdk/logic/Logic.java

```diff
@@ -158,6 +220,7 @@ public static VarintResult getUVarint(byte [] buffer, int bufferOffset) {
      * @return
      * @throws IOException
      */
+    @Deprecated
```

Comment:
@ahangsu What's your thinking on the following?

Since `checkProgram` and `readProgram` appear to be the most centrally exposed (public) methods, I'm thinking it'd be helpful to add a bit more context via a javadoc `@deprecated` tag (because Java 7 `@Deprecated` lacks fields) on _why_ it's deprecated + stand-in replacement methods.

I can be convinced to leave as is.  I'm hoping there's a lightweight 1-2 sentences that can be added in here rather than something more onerous.

Reply:
yes, the `@deprecated` tag in `java-docs` totally makes sense, as a detained explanation for the decision on deprecation, and this can also be extended to other SDKs.

Reply:
just added some example message under `@deprecation` tag.